great_expectations/expectations/core/expect_column_values_to_be_between.py

```diff
@@ -107,6 +109,60 @@ class ExpectColumnValuesToBeBetween(ColumnMapExpectation):
         "include_config": True,
         "catch_exceptions": False,
         "meta": None,
+        "auto": False,
+        "profiler_config": {
```

Comment:
Why are these dictionaries instead of typed profiler configs?

Reply:
@jcampbell The reasoning is the same as in the case of the other two expectations, now imbued with the default profiler.  Specifically, in an earlier conversation with @anthonyburdi we agreed that it would be better to defer the instantiation until needed (because the auto flag can be False, meaning that the user does not want the automatic expectation kwargs estimation feature to be turned on). Either will work -- the code would have to convert at some point in order to reconcile custom and default configurations.

Reply:
+1 on using real types here that can encapsulate the reconciliation logic and even pair down the default values substantially. For example, you can imagine not having to serialize any default values.

As it stands, this bucket of hardcoded strings will be impossible to maintain and a source for many future defects. Functionally what is happening here is amazing, but I'd rather move slowly and create something maintainable than adopt this pattern over the rest of our expectations.

Reply:
Reiterating here: While I think we should not instantiate a `Profiler` object here we should instantiate a `ProfilerConfig` object instead of a dictionary to take advantage of the validation, documentation and guiding parameters of the `ProfilerConfig` class. Alex and I discussed and this will be changed before merge.

Reply:
@jcampbell This got done in a better way to use typed configuration object (hat tip to @donaldheppner and @anthonyburdi ) as well as per your suggestion to define the default profiler above.  Thanks.